src/features/legend_panel/components/PanelContent/PanelContent.tsx

```diff
@@ -1,12 +1,25 @@
-import { useAtom } from '@reatom/react-v2';
+import { useAtom } from '@reatom/npm-react';
 import { useMemo } from 'react';
 import { mountedLayersAtom } from '~core/logical_layers/atoms/mountedLayers';
+import { FOCUSED_GEOMETRY_LOGICAL_LAYER_ID } from '~core/focused_geometry/constants';
+import { focusedGeometryAtom } from '~core/focused_geometry/model';
+import { referenceAreaAtom } from '~core/shared_state/referenceArea';
+import { REFERENCE_AREA_LOGICAL_LAYER_ID } from '~features/reference_area/constants';
 import { LegendsList } from '../LegendsList/LegendsList';
 import s from './PanelContent.module.css';
 
 export function PanelContent() {
-  const [layers] = useAtom(mountedLayersAtom);
-  const layersAtoms = useMemo(() => Array.from(layers.values()), [layers]);
+  const [layers] = useAtom(mountedLayersAtom.v3atom);
+  const [focusedGeometry] = useAtom(focusedGeometryAtom.v3atom);
+  const [referenceAreaGeometry] = useAtom(referenceAreaAtom);
+
+  const layersAtoms = useMemo(() => {
+    return Array.from(layers.values()).filter((layer) => {
+      if (layer.id === FOCUSED_GEOMETRY_LOGICAL_LAYER_ID) return focusedGeometry;
+      if (layer.id === REFERENCE_AREA_LOGICAL_LAYER_ID) return referenceAreaGeometry;
+      else return true;
```

Comment:
Can we somehow avoid the code duplication?

Reply:
Yeah, that's a good idea. We can avoid duplication by defining a filter util  function. Maybe you can help me and suggest, where I can store this helper, cause its used in 2 different features: `legend_panel` and `layers_panel`.

Reply:
Somewhere in the `core`, all common code is located in the `core` directory.